File: app/src/main/java/vn/loitp/app/activity/customviews/button/circularimageclick/CircularImageClickActivity.java
package vn.loitp.app.activity.customviews.button.circularimageclick;

import android.os.Bundle;
import android.view.View;

import com.core.base.BaseFontActivity;
import com.views.button.circularimageclick.LCircularClickImageButton;

import loitp.basemaster.R;

//guide https://github.com/hoang8f/android-flat-button
public class CircularImageClickActivity extends BaseFontActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((LCircularClickImageButton) findViewById(R.id.circleButton)).setOnCircleClickListener(new LCircularClickImageButton.OnCircleClickListener() {
            @Override
            public void onClick(View v) {
                showShort("onClick");
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_circular_image_click;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/button/circularimageclick/CircularImageClickActivity.kt
package vn.loitp.app.activity.customviews.button.circularimageclick

import android.os.Bundle
import android.view.View
import androidx.core.view.isVisible
import com.annotation.IsFullScreen
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LUIUtil
import com.views.button.circularimageclick.LCircularClickImageButton
import kotlinx.android.synthetic.main.activity_button_circular_image_click.*
import vn.loitp.app.R

@LogTag("CircularImageClickActivity")
@IsFullScreen(false)
class CircularImageClickActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_button_circular_image_click
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.viewShadow?.isVisible = true
            this.tvTitle?.text = CircularImageClickActivity::class.java.simpleName
        }
        circleButton.setOnCircleClickListener(object :
            LCircularClickImageButton.OnCircleClickListener {
            override fun onClick(v: View?) {
                showShortInformation("onClick")
            }
        })
    }
}
